conjur_api/http/api.py

```diff
@@ -84,17 +86,23 @@ def api_key(self) -> str:
     # pylint: disable=missing-docstring
     async def api_token(self) -> str:
         """
-        @return: Conjur api_token
+        @return: Conjur api_token based on the authentication method
         """
-        if not self._api_token or datetime.now() > self.api_token_expiration:
+        if self._api_token and datetime.now() < self.api_token_expiration:
+            logging.debug("Using cached API token...")
+            return self._api_token
+
+        if self._authentication_type == AuthnTypes.UsernamePassword:
             logging.debug("API token missing or expired. Fetching new one...")
             self.api_token_expiration = datetime.now() + timedelta(minutes=self.API_TOKEN_DURATION)
             self._api_token = await self.authenticate()
-
             return self._api_token
 
-        logging.debug("Using cached API token...")
-        return self._api_token
+        elif self._authentication_type == AuthnTypes.OIDC:
```

Comment:
I know there are only two enums but maybe 
```suggestion
        else:
```

Reply:
@mbenita-Cyberark  I think it's better to leave it with elif to emphasize there are potentially more than 2 options